### PR TITLE
Make std.math.nextUp and nextDown and nextafter work in CTFE for extended-precision real

### PR DESCRIPTION
**Pull request by n8sh**
PR #7176 didn't include extended-precision reals due to problems with FreeBSD in the CI suite. Opening this as a draft to see if I can get tests to pass.

### CODE REVIEW

File: std/math.d [hunk 1]
```diff
@@ -7082,15 +7120,21 @@ float nextDown(float x) @safe pure nothrow @nogc
 
 @safe pure nothrow @nogc unittest
 {
-    static if (floatTraits!(real).realFormat == RealFormat.ieeeExtended)
+    static if (floatTraits!(real).realFormat == RealFormat.ieeeExtended ||
+               floatTraits!(real).realFormat == RealFormat.ieeeDouble ||
+               floatTraits!(real).realFormat == RealFormat.ieeeExtended53 ||
+               floatTraits!(real).realFormat == RealFormat.ieeeQuadruple)
```
Comment (n8sh) [inline std/math.d:7126]: Let's see if we can enable the tests for all formats except `RealFormat.ibmExtended`.

**Comment by thewilsonator** [comment]
Is this good to go?

**Comment by n8sh** [comment]
Think so.

File: std/math.d (hunk 1 above)
Comment (ibuclaw) [reply std/math.d:7126]: @n8sh do you need a freebsd/x86 or aarch64 machine to test?

**Comment by ibuclaw** [comment]
Codecov is making this difficult to review in the UI.

File: std/math.d [hunk 2]
```diff
@@ -6883,6 +6883,44 @@ debug(UnitTest)
 real nextUp(real x) @trusted pure nothrow @nogc
 {
     alias F = floatTraits!(real);
+    static if (F.realFormat != RealFormat.ieeeDouble)
+    if (__ctfe)
```
Comment (ibuclaw) [inline std/math.d:6887]: Braces/indentation?

File: std/math.d (hunk 1 above)
Comment (n8sh) [reply std/math.d:7126]: @ibuclaw that would be useful

**Comment by n8sh** [comment]
>Codecov is making this difficult to review in the UI.

I don't disagree, on top of which the Codecov report is certainly wrong.

File: std/math.d (hunk 1 above)
Comment (ibuclaw) [reply std/math.d:7126]: Both pass ctfe tests without issue.

File: std/math.d [hunk 3]
```diff
@@ -6883,6 +6883,44 @@ debug(UnitTest)
...
+            return x;
+        real delta;
+        // Start with a decent estimate of delta.
+        if (x <= 0x1.ffffffffffffep+1023 && x >= -double.max)
+        {
+            const double d = cast(double) x;
+            static if (real.mant_dig >= 64)
+                delta = (cast(real) nextUp(d) - cast(real) d) * 0x1p-11L;
+            else // e.g. RealFormat.ieeeExtended53
+                delta = cast(real) nextUp(d) - cast(real) d;
```
Comment (ibuclaw) [inline std/math.d:6901]: Doesn't look like this is necessary for 80-bit reals with 53-bit precision, you can just use the first branch unconditionally.

**Review by ibuclaw — APPROVED** [review]

**Pull request merged**